src/model/element.ts

```diff
@@ -70,6 +72,7 @@ export class Element implements Feature {
   jsdoc?: jsdoc.Annotation;
   astNode: estree.Node|null;
   kinds: Set<string> = new Set(['element']);
+  warnings: Warning[] = [];
```

Comment:
What kind of feature can have warnings now? Does it make sense to add directly to the Feature class? Then all documents and/or elements could to have a `getWarnings()` method that returns all warnings contained within.

WDYT? I've been thinking about how to return all warnings found within a file, which is why this is top of mind.